azuredeploy.bicep

```diff
@@ -327,6 +985,230 @@ resource serverName_databaseName 'Microsoft.DBforMySQL/flexibleServers/databases
+resource avdAgentDscExtension 'Microsoft.Compute/virtualMachines/extensions@2018-10-01' = [for i in range(0, AVDnumberOfInstances): {
+  name: 'AvdAgentDSC'
+  parent: vm[i]
+  location: location
+  properties: {
+    publisher: 'Microsoft.Powershell'
+    type: 'DSC'
+    typeHandlerVersion: '2.73'
+    autoUpgradeMinorVersion: true
+    settings: {
+      modulesUrl: artifactsLocation
+      configurationFunction: 'Configuration.ps1\\AddSessionHost'
+      properties: {
+        hostPoolName: hostPool.name
+        registrationInfoToken: getHostPool.properties.registrationInfo.token
+        aadJoin: false
+      }
+    }
+  }
+  dependsOn: [
+    getHostPool
+  ]
+}]
+
+resource domainJoinExtension 'Microsoft.Compute/virtualMachines/extensions@2018-10-01' = [for i in range(0, AVDnumberOfInstances): {
+  name: 'DomainJoin'
+  parent: vm[i]
+  location: location
+  properties: {
+    publisher: 'Microsoft.Compute'
+    type: 'JsonADDomainExtension'
+    typeHandlerVersion: '1.3'
+    autoUpgradeMinorVersion: true
+    settings: {
+      name: adDomainFqdn
+      ouPath: adOuPath
+      user: domainJoinUsername
+      restart: 'true'
+      options: '3'
+    }
+    protectedSettings: {
+      password: domainJoinPassword
+    }
+  }
+  dependsOn: [
+    avdAgentDscExtension[i]
+  ]
+}]
+
+resource dependencyAgentExtension 'Microsoft.Compute/virtualMachines/extensions@2018-10-01' = [for i in range(0, AVDnumberOfInstances): {
+  name: 'DAExtension'
+  parent: vm[i]
+  location: location
+  properties: {
+    publisher: 'Microsoft.Azure.Monitoring.DependencyAgent'
+    type: 'DependencyAgentWindows'
+    typeHandlerVersion: '9.5'
+    autoUpgradeMinorVersion: true
+  }
+}]
+
+resource antiMalwareExtension 'Microsoft.Compute/virtualMachines/extensions@2018-10-01' = [for i in range(0, AVDnumberOfInstances): {
+  name: 'IaaSAntiMalware'
+  parent: vm[i]
+  location: location
+  properties: {
+    publisher: 'Microsoft.Azure.Security'
+    type: 'IaaSAntimalware'
+    typeHandlerVersion: '1.5'
+    autoUpgradeMinorVersion: true
+    settings: {
+      AntimalwareEnabled: true
+    }
+  }
+}]
+
+resource ansibleExtension 'Microsoft.Compute/virtualMachines/extensions@2018-10-01' = [for i in range(0, AVDnumberOfInstances): {
+  name: 'AnsibleWinRM'
+  parent: vm[i]
+  location: location
+  properties: {
+    publisher: 'Microsoft.Compute'
+    type: 'CustomScriptExtension'
+    typeHandlerVersion: '1.10'
+    autoUpgradeMinorVersion: true
+    settings: {
+      fileUris: ['https://raw.githubusercontent.com/ansible/ansible/devel/examples/scripts/ConfigureRemotingForAnsible.ps1']
+    }
+    protectedSettings: {
+      commandToExecute: 'powershell.exe -Command \'./ConfigureRemotingForAnsible.ps1; exit 0;\''
+    }
+  }
+}]
+
+
+
+
+
 output MySQLHostName string = '${uniqueServerName}.mysql.database.azure.com'
 output MySqlUserName string = '${administratorLogin}@${uniqueServerName}'
 output webSiteFQDN string = '${uniqueWebSiteName}.azurewebsites.net'
```

Comment:
The URL is a property of the App Service. I suggest using instead of hardcoding the domain name (which creates issues in other environments like Gov).